TerriaJS/terriajs
# HOC `withControlledVisibility` should extend propTypes of the wrapped component

### What this PR does

https://github.com/TerriaJS/terriajs/blob/9e6a18f8c656701f2cad8d550d32da468f8eccc0/lib/ReactViews/StandardUserInterface/processCustomElements.jsx#L71-L79

Here we check  `propTypes` to decide whether to forward `smallScreen` prop to the child components. But if the child component (like [MenuItem](https://github.com/TerriaJS/terriajs/blob/main/lib/ReactViews/StandardUserInterface/customizable/MenuItem.jsx)) is wrapped by `withControlledVisibility` then the check fails and the prop is discarded. This resulted in some buttons to not be styled correctly in mobile.

eg from #5782 

before:
![image](https://user-images.githubusercontent.com/1430/132297456-075600c8-85e7-420c-b2da-9a31a1ce859f.png)

after:
![image](https://user-images.githubusercontent.com/1430/132297516-b66081fe-6eff-44e1-9f4e-9caed33e7f34.png)

### Testing

- Open [link](http://ci.terria.io/extend-proptypes/)
- Switch to mobile view
- Open the hamburger menu
- The About link should have the same style as the rest


### Checklist
